src/App.tsx

```diff
@@ -17,18 +39,30 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                setSelectedStatus={setSelectedStatus}
+                setQuery={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length <= 0 ? (
+                <Loader />
```

Comment:
Show loader only in case you are loading something, if you get a response with an empty array you will show the loader infinitely